pkg/engine/apiPath.go

```diff
@@ -20,6 +21,12 @@ func NewAPIPath(path string) (*APIPath, error) {
 	trimmedPath := strings.Trim(path, "/ ")
 	paths := strings.Split(trimmedPath, "/")
 
+	if paths[0] == "apis" && len(paths) > 7 {
```

Comment:
Yes. I guess here deserves a comment to explain why we accept URLs whose length is greater than 7.
We can link to the issue and add the example URL.

Reply:
Yeah, this is WIP for now, I'd like to discuss this with other maintainers.
We'll have to add tests, docs, maybe improve naming, etc...

Reply:
any update?

Reply:
@JimBugwadia @chipzoller @realshuting WDYT ?

Reply:
Let's get this in 1.8.

Reply:
@eddycharly @chipzoller @samj1912 - lets complete and review the KDP (https://github.com/kyverno/KDP/issues/14). I see the value in this, but have concerns on the security model including the API authentication and authorization. Those can be addressed, but deserve some thought.

Reply:
@JimBugwadia this PR is only about in cluster kube-api-server calls, what on the security side could be dangerous ?

Reply:
@eddycharly - looks like I confused this with https://github.com/kyverno/kyverno/issues/3980. Kyverno calling the API server should be fine.

Reply:
It May be worth a description of why compared to length 7.

Reply:
TBH, I plan to remove all constraints in the next PR, IMHO there's no point trying to make sense of the requested URL here. WDYT ?

Reply:
@JimBugwadia may have a better idea on this.

Reply:
FWIW, removing constraints would fix https://github.com/kyverno/kyverno/issues/4239